test/subspace-test-service/src/mock.rs

```diff
@@ -147,22 +151,36 @@ impl MockPrimaryNode {
         self.next_slot
     }
 
-    /// Produce slot
-    pub fn produce_slot(&mut self) -> Slot {
+    /// Produce a slot and wait for the acknowledgement of all the slot notification subscribers
+    pub async fn produce_slot(&mut self) -> Slot {
```

Comment:
This implies a lot in fact, it won't resolve until the bundle is produced and submitted, let's make it more explicit.

```suggestion
    pub async fn produce_slot_and_wait_for_bundle_submission(&mut self) -> Slot {
```